cmd/simulator/load/awm_worker.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package load
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/cmd/simulator/txs"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var _ txs.Worker[*AwmTx] = &awmWorker{}
+
+type AwmTx struct {
+	AwmID common.Hash // a hash of the payload used to uniquely identify this awm message
+	Tx    *types.Transaction
+}
+
+func (a *AwmTx) Hash() common.Hash {
+	return a.Tx.Hash() // this is used by the batch worker to track tx times
+}
+
+type awmWorker struct {
+	worker      *singleAddressTxWorker
+	onIssued    func(common.Hash)
+	onConfirmed func(common.Hash)
+	onClosed    func()
+}
```

Comment:
This doesn't seem specific to awm. Could we just add these callbacks into `singleAddressWorker` ?